test/Tests/src/Data/Text/Regex_2_Spec.enso

```diff
@@ -1,199 +1,98 @@
-                match.at 4 . group 0 . should_equal "ij"
+        Test.specify "should return `[]` when an all match match fails" <|
+            pattern = Regex_2.compile "(aa)" []
+            input = "abcdefghij"
+            match = pattern.match_all input
+            match . should_equal []
 
-            Test.specify "should return `Nothing` when an all match match fails" <|
-                pattern = engine.compile "(aa)" []
-                input = "abcdefghij"
-                match = pattern.match input mode=Regex_Mode.All
-                match . should_equal Nothing
-
-            Test.specify "should be able to `match` the pattern against the entire input" <|
-                pattern = engine.compile "(.. .. )(?<letters>.+)()??(?<empty>)??" []
-                input = "aa ab abc a bc bcd"
-                match = pattern.match input mode=Regex_Mode.Full
-                match . should_be_a Default_Engine.Match.Value
-                match.text 0 . should_equal input
-
-            Test.specify "should return `Nothing` if a full match does not match the entire input" <|
-                pattern = engine.compile "(..)" []
-                input = "aa ab"
-                full_match = pattern.match input mode=Regex_Mode.Full
-                full_match . should_equal Nothing
-                match = pattern.match input mode=Matching_Mode.First
-                match . should_be_a Default_Engine.Match.Value
-
-            Test.specify "should be able to `match` the pattern against bounded input" <|
-                pattern = engine.compile "(..)" []
-                input = "abcdefghij"
-                match = pattern.match input mode=(Regex_Mode.Bounded 2 8)
-                match.length . should_equal 3
-                match.at 0 . text 0 . should_equal "cd"
-                match.at 1 . text 0 . should_equal "ef"
-                match.at 2 . text 0 . should_equal "gh"
-
-            Test.specify "should correctly handle empty patterns" pending="Figure out how to make Regex correctly handle empty patterns." <|
-               pattern = engine.compile "" []
-               match_1 = pattern.match "" mode=Regex_Mode.All
-               match_1.length . should_equal 1
-               match_1.at 0 . start 0 . should_equal 0
-               match_1.at 0 . end 0 . should_equal 0
-
-               match_2 = pattern.match "ABC" mode=Regex_Mode.All
-               match_2.length . should_equal 4
-               match_2.at 0 . start 0 . should_equal 0
-               match_2.at 0 . end 0 . should_equal 0
-               match_2.at 1 . start 0 . should_equal 1
-               match_2.at 1 . end 0 . should_equal 1
-               match_2.at 3 . start 0 . should_equal 3
-               match_2.at 3 . end 0 . should_equal 3
-
-    Test.group "The default regex engine's Pattern.find" <|
+        Test.specify "should correctly handle empty patterns" pending="Figure out how to make Regex correctly handle empty patterns." <|
```

Comment:
```suggestion
        Test.specify "should correctly handle empty patterns" <|
```

Can we maybe try running these tests again? Seems like the new Regexp implementation should take this requirement into account.